internal/core/hostid/azure.go

```diff
@@ -49,5 +53,19 @@ func AzureUniqueID() string {
 		return ""
 	}
 
-	return fmt.Sprintf("%s/%s/microsoft.compute/virtualmachines/%s", compute.Doc.SubscriptionID, compute.Doc.ResourceGroupName, compute.Doc.Name)
+	if compute.Doc.VMScaleSetName == "" {
+		return fmt.Sprintf("%s/%s/microsoft.compute/virtualmachines/%s", compute.Doc.SubscriptionID, compute.Doc.ResourceGroupName, compute.Doc.Name)
+	}
+
+	parsedName := strings.Split(compute.Doc.Name, "_")
+
+	// names of VM's in VMScalesets seem to follow the of `<scale-set-name>_<instance-id>`
+	// where scale-set-name is alphanumeric (and is the same as compute.vmScaleSetName
+	// field from the metadata endpoint)
+	if len(parsedName) != 2 {
+		return ""
```

Comment:
Can you just do `strings.TrimLeft(compute.Doc.VMScaleSetName + "_")` instead?  It seems a bit flaky to fail out with no azure_resource_id if the name isn't exactly as you expect it with a Scale Set.

Reply:
Updated to use `TrimLeft`